generative/3d_ldm/inference.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import logging
+from datetime import datetime
+import nibabel as nib
+from pathlib import Path
+
+import os
+import sys
+import numpy as np
+
+import torch
+from generative.inferers import LatentDiffusionInferer
+from generative.networks.schedulers import DDPMScheduler
+from monai.config import print_config
+from monai.utils import set_determinism
+
+from utils import define_instance
+
+
+def main():
+    parser = argparse.ArgumentParser(description="PyTorch Latent Diffusion Model Inference")
+    parser.add_argument(
+        "-e",
+        "--environment-file",
+        default="./config/environment.json",
+        help="environment json file that stores environment path",
+    )
+    parser.add_argument(
+        "-c",
+        "--config-file",
+        default="./config/config_train_48g.json",
+        help="config json file that stores hyper-parameters",
+    )
+    parser.add_argument(
+        "-n",
+        "--num",
+        type=int,
+        default=1,
+        help="number of generated images",
+    )
+    args = parser.parse_args()
+
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+    print_config()
+    torch.backends.cudnn.benchmark = True
+    torch.set_num_threads(4)
+
+    env_dict = json.load(open(args.environment_file, "r"))
+    config_dict = json.load(open(args.config_file, "r"))
+
+    for k, v in env_dict.items():
+        setattr(args, k, v)
+    for k, v in config_dict.items():
+        setattr(args, k, v)
+
+    directory = os.environ.get("MONAI_DATA_DIRECTORY")
```

Comment:
Would be good to change to

```

import tempfile
monai_directory = os.environ.get("MONAI_DATA_DIRECTORY")
directory= tempfile.mkdtemp() if monai_directory is None else monai_directory 
print(directory)
```

Reply:
This parameter is unused. I personally do not like the idea of hardcoding data path to main script... Do we have to do this?